src/pages/profile.tsx

```diff
@@ -22,24 +22,28 @@ import useUserContext from 'useUserContext';
 const UserProfile: NextPageWithLayout = () => {
   const [user, setUser] = useUserContext();
   const [name, setName] = useState<string>('');
-  const [show, setShow] = useState(false);
+  const [loaded, setLoaded] = useState(true);
 
   useEffect(() => {
     setName(user.name || '')
   }, [user]);
 
   const handleSubmit = async () => {
+    setLoaded(false);
+
     if (name) {
       const updatedUser = structuredClone(user);
       updatedUser.name = name;
 
       // TODO: Handle error display globally. Redact server-side errors.
       try {
         await trpc.users.update.mutate(updatedUser);
+        toast.success("个人信息已保存")
         setUser(updatedUser);
-        setShow(!show);
+        setLoaded(true);
       } catch(e) {
         toast.error((e as Error).message);
```

Comment:
We recently implemented global error displays so we can remove this line. Also, change this `catch` block to using `finally` instead:

```
finally {
  setLoading(false);
}
```

Reply:
done